test/syslogger_SUITE.erl

```diff
@@ -0,0 +1,151 @@
+-module(syslogger_SUITE).
+
+-export([all/0,
+         init_per_suite/1,
+         end_per_suite/1,
+         init_per_testcase/2,
+         end_per_testcase/2]).
+
+-export([
+    syslogger_is_used_to_log/1,
+    log_at_every_level/1,
+    log_at_custom_level/1
+]).
+
+-include_lib("common_test/include/ct.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+-define(HANDLER_ID, sys_log).
+-define(IDENT, "mongooseim").
+-define(FACILITY, local0).
+
+all() ->
+    [syslogger_is_used_to_log,
+     log_at_every_level,
+     log_at_custom_level
+    ].
+
+init_per_suite(Config) ->
+    LoggerConfig = logger:get_primary_config(),
+    logger:set_primary_config(level, info),
+
+    setup_meck(Config),
+    ok = mongoose_logger_running(),
+
+    [{logger_primary_config, LoggerConfig} | Config].
+
+end_per_suite(Config) ->
+    logger:remove_handler(?HANDLER_ID),
+    logger:set_primary_config(?config(logger_primary_config, Config)),
+
+    unload_meck(Config),
+    ok.
+
+init_per_testcase(_TC, Config) ->
+    meck:reset(syslogger),
+    Config.
+
+end_per_testcase(_TC, _Config) ->
+    ok.
+
+%%
+%% Tests
+%%
+syslogger_is_used_to_log(_Config) ->
+    logger:log(error, "Something"),
+
+    ?assert(meck:validate(syslogger)),
+    ok = meck:wait(syslogger, log, [#{level => error,
+                                           msg => {string, "Something"},
+                                           meta => '_'},
+        #{id => ?HANDLER_ID, ident => ?IDENT, facility => ?FACILITY}],
+        timer:seconds(1)).
```

Comment:
This indentation is strange, it doesn't make it obvious that the second map is part of the same list than the first map, and that this same second map and the `timer` call are also parameters of `meck:wait/4`. Perhaps something like...
```suggestion
    ok = meck:wait(syslogger, log,
                              [#{level => error, msg => {string, "Something"}, meta => '_'},
                               #{id => ?HANDLER_ID, ident => ?IDENT, facility => ?FACILITY}],
                              timer:seconds(1)).
```
or
```suggestion
    ok = meck:wait(syslogger, log, [#{level => error, msg => {string, "Something"}, meta => '_'},
                                                        #{id => ?HANDLER_ID, ident => ?IDENT, facility => ?FACILITY}],
                              timer:seconds(1)).
```
(The second is more readable but too close to being a long line... it's all about aesthetics so you decide 😄

Reply:
Missed this, thanks!